packages/api/cms-api/src/redirects/redirects.service.ts

```diff
@@ -1,22 +1,31 @@
-import { FilterQuery } from "@mikro-orm/core";
+import { EntityRepository, FilterQuery } from "@mikro-orm/core";
 import { InjectRepository } from "@mikro-orm/nestjs";
-import { EntityRepository } from "@mikro-orm/postgresql";
 import { forwardRef, Inject, Injectable } from "@nestjs/common";
 
 import { PageTreeService } from "../page-tree/page-tree.service";
 import { PageTreeNodeInterface } from "../page-tree/types";
-import { Redirect } from "./entities/redirect.entity";
-import { RedirectGenerationType, RedirectSourceTypeValues, RedirectTargetTypeValues } from "./redirects.enum";
+import { RedirectInterface } from "./entities/redirect-entity.factory";
+import { RedirectGenerationType, RedirectSourceTypeValues } from "./redirects.enum";
+import { REDIRECTS_LINK_BLOCK, RedirectsLinkBlock } from "./redirects.module";
 
 @Injectable()
 export class RedirectsService {
     constructor(
-        @InjectRepository(Redirect) private readonly repository: EntityRepository<Redirect>,
+        @InjectRepository("Redirect") private readonly repository: EntityRepository<RedirectInterface>,
         @Inject(forwardRef(() => PageTreeService)) private readonly pageTreeService: PageTreeService,
+        @Inject(REDIRECTS_LINK_BLOCK) private readonly linkBlock: RedirectsLinkBlock,
```

Comment:
Is that preferred over a factory?

Reply:
Well, I haven't thought about creating the service via a factory at all. However, it is more difficult to inject a service created via a factory into another service. As the redirects service is used by the page tree service, I believe it's better to inject the link block instead.